Core/Src/bms.c

```diff
@@ -0,0 +1,56 @@
+#include "bms.h"
+#include "timer.h"
+#include "fault.h"
+#include "can.h"
+#include "cerberus_conf.h"
+#include <assert.h>
+#include <stdlib.h>
+
+#define BMS_CAN_MONITOR_DURATION 100 /*Duration of between petting bms monitor watchdog (in ticks)*/
+#define CAN_QUEUE_SIZE 5
+
+osMessageQueueId_t bms_monitor_queue;
+
+void bms_fault_callback();
+
+osThreadId_t bms_monitor_handle;
+const osThreadAttr_t bms_monitor_attributes = {
+	.name = "BMSCANMonitor",
+	.stack_size = 128 * 8,
+	.priority = (osPriority_t)osPriorityLow1 /*TO-DO: Adjust priority*/
+};
+
+void bms_fault_callback() {
+	fault_data_t fault_data = { .id = BMS_CAN_MONITOR_FAULT, .severity = DEFCON1 }; /*TO-DO: update severity*/
+
+	fault_data.diag = "Failing To Receive CAN Messages from Shepherd";
+	queue_fault(&fault_data);
+}
+
+bms_t* bms_init() {
+
+	bms_t* bms = malloc(sizeof(bms_t));
+	assert(bms);
+
+	/*TO-DO: specify timer attributes*/
+	bms->bms_monitor_timer = osTimerNew(&bms_fault_callback, osTimerPeriodic, NULL, NULL);  
```

Comment:
Could be wrong but I don't think should be periodic, we want to be continuously resetting a timer to only sets off once. Check here on a sample timer implementation I did for steering wheel button debouncing: https://github.com/Northeastern-Electric-Racing/Cerberus/blob/9e6cc02537e24feeff18ae8dbcb41a1a27363a14/Core/Src/steeringio.c#L34C60-L34C71